PWGJE/Tasks/bjetTreeCreator.cxx

```diff
@@ -0,0 +1,634 @@
+        bjetSVParamsTable(bjetParamsTable.lastIndex() + 1, candSV.pt(), deltaRJetSV, massSV, energySV / myJet.energy(), candSV.impactParameterXY(), candSV.cpa(), candSV.chi2PCA(), candSV.decayLengthXY(), candSV.errorDecayLengthXY(), candSV.decayLength(), candSV.errorDecayLength());
+      }
+      svIndices.push_back(bjetSVParamsTable.lastIndex());
+
+      registry.fill(HIST("h2_LxyS_jetpT"), myJet.pt(), candSV.decayLengthXY() / candSV.errorDecayLengthXY(), eventweight);
+      registry.fill(HIST("h2_Dispersion_jetpT"), myJet.pt(), candSV.chi2PCA(), eventweight);
+      registry.fill(HIST("h2_SVMass_jetpT"), myJet.pt(), massSV, eventweight);
+
+      if (doprocessMCJets) {
+        if (jetFlavor == 2) {
+          registry.fill(HIST("h2_LxyS_jetpT_bjet"), myJet.pt(), candSV.decayLengthXY() / candSV.errorDecayLengthXY(), eventweight);
+          registry.fill(HIST("h2_Dispersion_jetpT_bjet"), myJet.pt(), candSV.chi2PCA(), eventweight);
+          registry.fill(HIST("h2_SVMass_jetpT_bjet"), myJet.pt(), massSV, eventweight);
+        } else if (jetFlavor == 1) {
+          registry.fill(HIST("h2_LxyS_jetpT_cjet"), myJet.pt(), candSV.decayLengthXY() / candSV.errorDecayLengthXY(), eventweight);
+          registry.fill(HIST("h2_Dispersion_jetpT_cjet"), myJet.pt(), candSV.chi2PCA(), eventweight);
+          registry.fill(HIST("h2_SVMass_jetpT_cjet"), myJet.pt(), massSV, eventweight);
+        } else {
+          registry.fill(HIST("h2_LxyS_jetpT_lfjet"), myJet.pt(), candSV.decayLengthXY() / candSV.errorDecayLengthXY(), eventweight);
+          registry.fill(HIST("h2_Dispersion_jetpT_lfjet"), myJet.pt(), candSV.chi2PCA(), eventweight);
+          registry.fill(HIST("h2_SVMass_jetpT_lfjet"), myJet.pt(), massSV, eventweight);
+        }
+      }
+    }
+  }
+
+  template <typename AnyCollision, typename AnalysisJet, typename AnyTracks, typename SecondaryVertices>
+  void processJetTrackInfo(AnyCollision const& collision, AnalysisJet const& analysisJet, AnyTracks const& allTracks, SecondaryVertices const& allSVs, std::vector<int>& trackIndices, int jetFlavor = 0, double eventweight = 1.0)
+  {
+    jetConstituents.clear();
+
+    for (auto& jconstituent : analysisJet.template tracks_as<AnyTracks>()) {
+      fastjetutilities::fillTracks(jconstituent, jetConstituents, jconstituent.globalIndex());
+
+      if (jconstituent.pt() < trackPtMin) {
+        continue;
+      }
+
+      auto constituent = jconstituent.template track_as<OriginalTracks>();
+      double deltaRJetTrack = jetutilities::deltaR(analysisJet, constituent);
+      double dotProduct = RecoDecay::dotProd(std::array<float, 3>{analysisJet.px(), analysisJet.py(), analysisJet.pz()}, std::array<float, 3>{constituent.px(), constituent.py(), constituent.pz()});
+      int sign = jettaggingutilities::getGeoSign(collision, analysisJet, constituent);
+
+      float RClosestSV = 10.;
+      for (const auto& candSV : analysisJet.template secondaryVertices_as<SecondaryVertices>()) {
```

Comment:
It is possible. But here we are looping over all the SVs insides a jet to see the distance between the track and the closest SV. If this feature is too week it will be removed.
I also think filling the SV table here will make a bit of a mess. So I think it's better to have them in separate functions.